crates/dataflow-jit/src/ir/literal.rs

```diff
@@ -73,9 +84,76 @@ impl StreamCollection {
         }
     }
 
+    pub const fn empty(layout: StreamLayout) -> StreamCollection {
+        match layout {
+            StreamLayout::Set(_) => Self::Set(Vec::new()),
+            StreamLayout::Map(..) => Self::Map(Vec::new()),
+        }
+    }
+
     pub fn is_empty(&self) -> bool {
         self.len() == 0
     }
+
+    pub fn consolidate(&mut self) {
+        match self {
```

Comment:
let's see if this has problems with NaN.

Reply:
It shouldn't, it calls `Constant`'s comparison impl